st2common/st2common/models/utils/action_alias_utils.py

```diff
@@ -117,36 +132,69 @@ def get_extracted_param_value(self):
         if not search_regex_tokens(((AT, AT_END), (AT, AT_END_STRING)), reg_tokens, backwards=True):
             reg = reg + r'\s*$'
 
-        # 3. Matching the command against our regex to get the param values
-        matched_stream = re.match(reg, param_stream, re.DOTALL)
+        return re.compile(reg, re.DOTALL)
 
+    def match_params_in_stream(self, matched_stream):
         if not matched_stream:
             # If no match is found we throw since this indicates provided user string (command)
             # didn't match the provided format string
             raise ParseException('Command "%s" doesn\'t match format string "%s"' %
-                                 (self._param_stream, self._format))
+                                 (self._original_param_stream, self._format))
 
         # Compiling results from the steps 1-3.
         if matched_stream:
             result = matched_stream.groupdict()
 
-        for param in optional:
+        # Apply optional parameters/add the default parameters
+        for param in self._optional:
             matched_value = result[param[0]] if matched_stream else None
             matched_result = matched_value or ''.join(param[1:])
             if matched_result is not None:
                 result[param[0]] = matched_result
 
-        if has_ending_pairs:
-            for pair in kv_pairs:
-                result[pair[0]] = ''.join(pair[2:])
+        # Apply given parameters
+        for pair in self._kv_pairs:
+            result[pair[0]] = ''.join(pair[2:])
 
         if self._format and not (self._param_stream.strip() or any(result.values())):
             raise ParseException('No value supplied and no default value found.')
 
         return result
 
+    def get_extracted_param_value(self):
+        """
+        Match command against the format string and extract paramters from the command string.
```

Comment:
Small type `paramters` -> `parameters`.

Reply:
Not the only place I misspell it. I'll fix it.